ui/components/MesheryFilters/FiltersCard.js

```diff
@@ -0,0 +1,301 @@
+//@ts-check
+import React, { useState } from "react";
+import {
+  Dialog, DialogActions, DialogContent, DialogTitle, Divider, Grid, IconButton, Tooltip, Typography
+} from "@material-ui/core";
+import Button from "@material-ui/core/Button";
+import DeleteIcon from "@material-ui/icons/Delete";
+import Fullscreen from "@material-ui/icons/Fullscreen";
+import Moment from "react-moment";
+import FlipCard from "../FlipCard";
+import { makeStyles } from "@material-ui/core/styles";
+import { UnControlled as CodeMirror } from "react-codemirror2";
+import FullscreenExit from "@material-ui/icons/FullscreenExit";
+import DoneAllIcon from '@material-ui/icons/DoneAll';
+
+const INITIAL_GRID_SIZE = { xl : 4, md : 6, xs : 12 };
+
+const useStyles= makeStyles(() => ({
+  cardButtons : {
+    display : "flex",
+    justifyContent : "flex-end",
+    alignItems : "center",
+    marginTop : "50px"
+  },
+  testsButton : {
+    marginRight : "0.5rem"
+  },
+  perfResultsContainer : {
+    marginTop : "0.5rem"
+  },
+  backGrid : {
+    marginBottom : "0.25rem",
+    minHeight : "6rem",
+    position : "relative"
+  },
+  deleteButton : {
+    width : "fit-content",
+    margin : "10 0 0 auto",
+    position : "absolute",
+    right : 0,
+    bottom : 0,
+  },
+  yamlDialogTitle : {
+    display : "flex",
+    alignItems : "center"
+  },
+  yamlDialogTitleText : {
+    flexGrow : 1
+  },
+  fullScreenCodeMirror : {
+    height : '100%',
+    width : '100%',
+    '& .CodeMirror' : {
+      minHeight : "300px",
+      height : '100%',
+      width : '100%'
+    }
+  },
+  maximizeButton : {
+    width : "fit-content",
+    margin : "0 0 0 auto",
+    position : "absolute",
+    right : 0,
+    top : 0
+  },
+  noOfResultsContainer : {
+    margin : "0 0 1rem",
+    '& div' : {
+      display : "flex",
+      alignItems : "center"
+    },
+  },
+  bottomPart : {
+    display : "flex",
+    justifyContent : "flex-end",
+    alignItems : "center",
+  },
+  lastRunText : {
+    marginRight : "0.5rem"
+
+  },
+  iconPatt : {
+    width : "24px",
+    height : "24px",
+    marginRight : "5px"
+  }
+}))
+
+//Full screen Dialog: Similar to the dialog on the table view, with few modification on parameters
+const YAMLDialog = ({
```

Comment:
YAML Dialog was also made universal when the component was created. It will be a great addition if we could use that thought.

Reply:
@Abhishek-kumar09 Thanks for the review, I've updated it (made components globally reusable)
This YAMLDialog doesn't seem universal, we could make this also universal now also. 

There are multiple items across configurations/lifecycles and performance which could be universal. V0.7 Refactoring will make sure of these approaches.

Reply:
@Chadha93 there are any number of these items that I think that we should address now, so that the lift to MUIv5 isn't laden with a bunch of need for restructuring and fixing, but is more about an upgrade from v4 to v5, so that we can work on this new version and iteratively redesign from there.

Reply:
What's different in Yaml dialog? I see it it taking general handlers for submitting etc and the text for yaml and all. Pointing out the specific problem will make us go deeper into its successful implementation.

Reply:
@leecalcote @Abhishek-kumar09 Yes the above points make sense, updated the component.

Reply:
@Abhishek-kumar09 Made the YAMLDialog universal. There is a minor issue when YAMLDialog is opened in full screen it is showing update for filters which we don't want. (Since the YAMLDialog is now a universal component we might need to add conditions to show the update button only for designs not on filters ) Please do checkout and test the pr.